src/classes/RD2_ScheduleService.cls

```diff
@@ -619,7 +619,26 @@ public without sharing class RD2_ScheduleService {
             isCurrent =             rdSchedule.StartDate__c <= RD2_ScheduleService.currentDate ? true : false;
             campaignId =            rdSchedule.Campaign__c;
             campaignName =          rdSchedule.Campaign__r.Name;
-            dayOfMonth =            rdSchedule.InstallmentPeriod__c == RD2_Constants.INSTALLMENT_PERIOD_MONTHLY ? rdSchedule.DayOfMonth__c : null;
+            if (rdSchedule.InstallmentPeriod__c == RD2_Constants.INSTALLMENT_PERIOD_MONTHLY) {
```

Comment:
I think Day of Month display conversion belongs in the visualizer and we can leave the schedule service as is.
Also, we can extract code into a method that will determine the Day of Month display value. So, if the method is added into the visualizer `Schedule`, it would be as below. Also, we need a test in the visualizer test class where Day of Month = Last_Day since current tests pass w/o this change.
```
public Schedule(Integer num, RD2_ScheduleService.ActiveSchedule schedule) {
            this.scheduleNumber = num;
            this.isCurrent = schedule.isCurrent;
            this.amount = schedule.installmentAmount;
            this.paymentMethod = schedule.paymentMethod;
            this.campaign = schedule.campaignName;
            this.startDate = schedule.startDate;
            this.endDate = schedule.endDate;
            this.period = schedule.installmentPeriod;
            this.frequency = schedule.installmentFrequency;
            this.dayOfMonth = determineDayOfMonth(schedule);
        }

        /**
         * @description Displays Day of Month picklist values 
         * @param schedule RD Schedule Service active schedule
         * @return String
         */
        private String determineDayOfMonth(RD2_ScheduleService.ActiveSchedule schedule) {
            if (schedule.installmentPeriod != RD2_Constants.INSTALLMENT_PERIOD_MONTHLY 
                || schedule.dayOfMonth == null
            ) {
                return null;
            }
            
            if (schedule.dayOfMonth == RD2_Constants.DAY_OF_MONTH_LAST_DAY) {                
                for (Schema.PicklistEntry picklistValue : RecurringDonationSchedule__c.DayOfMonth__c.getDescribe().getPicklistValues()) {
                    if (picklistValue.getValue() == RD2_Constants.DAY_OF_MONTH_LAST_DAY) {
                        return picklistValue.getLabel();
                    }
                }
            }

            return schedule.dayOfMonth;
        }

```

Reply:
Not sure I agree with this.  I'm open to hearing your thoughts about the logic, but I don't understand why the schedule service should continue filling the data structure with Last_Day.  

That is a value that is never usable by a consumer of this data, so it seems incorrect to me to allow it to be populated with an unusable value.

Reply:
The `RD2_ScheduleService.ActiveSchedule` is currently used by the visualizer, so there is no impact. If there was another class consuming the service method, then using the label instead of the API value would produce incorrect results. The reasoning was to keep the schedule service use the API value while the visualizer responsible for data display would use the label.

Reply:
So the assumption is that the visualizer will remain the only consumer.

Reply:
The assumption is the visualizer will not be the only consumer. Visualizer needs to output the label but not the schedule service.

Reply:
Trying to catch up on this. If this method is what is used to generate the Schedule data to render in the LWC UI, then we **would** want DayOfMonth to use the Label vs. the ApiName. If there's concern that the consumer of this can be expanded in the future, then we can rename this property to `dayOfMonthLabel` so it's clear what the usage is. I suspect this is going to come out of [W-040523](https://foundation--agf.na70.visual.force.com/a2x1E000001HuEkQAK) anyway. My question is why is this changed here as part of this WI?

Reply:
Okay, good news/bad news.  @force2b and I just discussed and we will defer the change to the Last Day data to be addressed in [W-040523](https://foundation--agf.na70.visual.force.com/a2x1E000001HuEkQAK).  That's the good news.  <joke>The bad new is I still think this belongs in the schedule service.</joke>